### PR TITLE
Adding unit tests & coverage report.

### PR DESCRIPTION
**Pull request by dread-pirate-johnny-spaceboots**
Am willing to write tests for the codebase, but I notice there's nothing setup here yet. I checked your other repos looking for tests to see how you usually go about it, but didn't find any. So I've setup MSTest and used the VS2019 extension "Fine Code Coverage" to generate a coverage report.:

image

I've only written a very basic test as a means of getting your thoughts on the approach.

### CODE REVIEW

File: Z80Bench.cs [hunk 1]
```diff
@@ -6,9 +6,9 @@
 
 namespace z80bench
 {
-    class Z80Bench
+    public class Z80Bench
```
Comment (johnny-spaceboots) [inline Z80Bench.cs:9]: Had to loosen up the default visibility of your main class/method to call it from the tests, short of reflecting anyway.

File: tests/z80benchtests/CLITests.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    public class CLITests {
+        private static readonly string[] _expectedErrorOutput = {
+            "--execute <offset>",
+            "--stack-pointer <offset>",
+            "--max-cycles <count>",
+            "--vram-compare filename[@offset]]"
+        };
+
+        [TestMethod]
+        public void ExitsWithErrorWithoutArguments() {
```
Comment (johnny-spaceboots) [inline tests/z80benchtests/CLITests.cs:17]: Simple test that asserts that when called with no args:
- The program returns -1
- The first line written to the console should be the usage: line
- All of the options should be present somewhere in the output (no assertion around their description or order)

File: tests/z80benchtests/z80benchtests.csproj [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.7.1" />
```
Comment (johnny-spaceboots) [inline tests/z80benchtests/z80benchtests.csproj:10]: Don't have a huge amount of C# exp, so defacto standards elude me, not sure if Microsofts default of MSTest is the way to go, but seems to work well enough

File: z80bench.sln [hunk 4]
```diff
@@ -5,6 +5,8 @@ VisualStudioVersion = 16.0.30225.117
 MinimumVisualStudioVersion = 10.0.40219.1
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "z80bench", "z80bench.csproj", "{E3BBA7C0-A604-47BE-A359-40D997596893}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "z80benchtests", "tests\z80benchtests\z80benchtests.csproj", "{0549E698-82B3-40B2-900F-1BC987D94739}"
```
Comment (johnny-spaceboots) [inline z80bench.sln:8]: Wasn't really sure what to call the test project so I just tacked 'tests' onto the name of the existing project ¯\_(ツ)_/¯

**Comment by maxim-zhao** [comment]
Hey, this is really cool, thanks. Yeah, my usual approach is to not unit test, although part of the point of this project is to help make it easier to test Z80 code. I think it would be good to put the testing and coverage reports into the CI (currently appveyor), what do you think?

**Comment by johnny-spaceboots** [comment]
> Hey, this is really cool, thanks. Yeah, my usual approach is to not unit test, although part of the point of this project is to help make it easier to test Z80 code. I think it would be good to put the testing and coverage reports into the CI (currently appveyor), what do you think?

For sure integrating them into the CI env is the way to go. Haven't used appveyor before, but I've used a handful of others, they all tend to be pretty straightforward. I'll get it integrated and update this PR with it, will let you know if I need any further details.

**Comment by johnny-spaceboots** [comment]
https://www.appveyor.com/docs/running-tests/#visual-studio-unit-testing-framework  Makes it sound like they auto-run the tests for certain setups (VS+MSTest being one of them). So this might already be working? If not it should just be a line or two in the .yaml config to specify the test assemblies.

**Comment by johnny-spaceboots** [comment]
Could also integrate CodeCov if you'd like, you'd need to setup a free account for it, but it integrates with AppVeyor and can display code coverage effects of PRs, code coverage badge on the readme page, etc.

https://about.codecov.io/product/features/
https://github.com/ceddlyburge/codecov-on-appveyor

**Comment by maxim-zhao** [comment]
Good news: indeed it did seem to do that: https://ci.appveyor.com/project/maxim-zhao/z80bench/builds/36880448 but then it reports no test results...

**Comment by maxim-zhao** [comment]
This seems to be the fix: https://stackoverflow.com/questions/48235374

**Comment by johnny-spaceboots** [comment]
Alright, amended the commit to contain the additional package and updated the appveyor yaml to contain the necessary testing line. 🤞

**Comment by johnny-spaceboots** [comment]
image

I think we got it.

**Comment by maxim-zhao** [comment]
Let's merge it and iterate from there. I may tweak things to be a bit more C#y.

**Comment by johnny-spaceboots** [comment]
> I may tweak things to be a bit more C#y.

Yes, please. I'd love to see what that'd entail. Also feel free to point out such things in any subsequent PR, I'm very receptive to feedback.

**Pull request merged**